src/zc/buildout/testing.py

```diff
@@ -110,12 +110,20 @@ def clean_up_pyc(*path):
 MUST_CLOSE_FDS = not sys.platform.startswith('win')
 
 def system(command, input='', with_exit_code=False):
+    env = dict(os.environ)
+    if 'xterm' in env.get('TERM', ''):
+        # xterm and its variants add invisible control characters,
+        # which we do not want as they mess up doctests.  See:
+        # https://github.com/buildout/buildout/pull/311
+        # http://bugs.python.org/issue19884
+        env['TERM'] = 'dumb'
```

Comment:
Seems like it would be fine on Windows. At least cygwin used to set it to dumb: http://kizzx2.com/blog/index.php/2009/01/07/cygwin-warning-terminal-is-not-fully-functional/
That then showed a warning when you do `git diff`, saying the terminal is not fully functional, but I have the same when I try that on OSX.
Other links possibly of interest:
http://stackoverflow.com/questions/28415681/how-to-set-term-environment-variable-in-windows-so-that-it-works-with-bash-via-s
http://stackoverflow.com/questions/7949956/why-does-git-diff-on-windows-warn-that-the-terminal-is-not-fully-functional
Note that on OSX when I say `export TERM=asldkjfasdlfaskdfjha` all still seems fine. :-) So unknown TERM settings don't usually completely break anything.

But in the end it is probably easier to use the same TERM setting for everyone.
Also: this is only in `testing.py` so it is not used when running buildout normally.
I will update the pull request.